backend/src/test/java/pl/edu/agh/niebieskiekotki/groups/GroupGeneratorTests.java

```diff
@@ -0,0 +1,73 @@
+package pl.edu.agh.niebieskiekotki.groups;
+
+import org.junit.jupiter.api.Test;
+import pl.edu.agh.niebieskiekotki.entitites.Student;
+import pl.edu.agh.niebieskiekotki.entitites.Term;
+import pl.edu.agh.niebieskiekotki.utility.Days;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class GroupGeneratorTests {
+
+    GroupGenerator generator = new GroupGenerator();
+
+    @Test
+    public void test(){
+        Student s1 = new Student();
+        Student s2 = new Student();
+        Student s3 = new Student();
+        Student s4 = new Student();
+        Student s5 = new Student();
+        Student s6 = new Student();
+        Student s7 = new Student();
+        Student s8 = new Student();
+
+        Term t1 = new Term(1L, Days.Monday, 1, null);
+        Term t2 = new Term(2L, Days.Tuesday, 2, null);
+        Term t3 = new Term(3L, Days.Wednesday, 3, null);
+        Term t4 = new Term(4L, Days.Thursday, 4, null);
+        Term t5 = new Term(5L, Days.Friday, 5, null);
+        Term t6 = new Term(6L, Days.Monday, 6, null);
+        Term t7 = new Term(7L, Days.Thursday, 3, null);
+
+        Map<Student, List<Term>> studentTerms = new HashMap<>();
+        studentTerms.put(s1, List.of(t1, t2, t3));
+        studentTerms.put(s2, List.of(t2, t3, t5));
+        studentTerms.put(s3, List.of(t3));
+        studentTerms.put(s4, List.of(t1, t5, t6));
+        studentTerms.put(s5, List.of(t3, t4));
+        studentTerms.put(s6, List.of(t3));
+        studentTerms.put(s7, List.of(t1, t2, t3, t4, t5, t6));
+        studentTerms.put(s8, List.of(t4, t5, t7));
+
```

Comment:
Dodałbym funkcję w stylu "testHelper" do której przekopiowałbym cały kod poniżej, Ułatwiło by to robienie większej ilości testów sprawdzających działanie algorytmu dla różnych przypadków.
Czyli np 
public void testHelper(Map<Student, List<Term>> studentTerms, List<Student> students)
bez adnotacji @Test które byłoby wywołane z tego miejsca w kodzie.